Assets/Scripts/Achievement/AchievementManager.cs

```diff
@@ -13,73 +15,113 @@ public class AchievementManager : Singleton<AchievementManager>
-            }
+                initialized = true;
+                Manager.GameManager.Instance.GameStarted += InvokeDefaultEvents;
+                if (achievementScriptable != null)
+                {
+                    achievementList = new List<Achievement>();
+
+                    for (int i = 0; i < achievementScriptable.achievementList.Count; i++)
+                    {
+                        Achievement achievement = new Achievement();
+                        achievement = achievementScriptable.achievementList[i];
+                        bool unlocked = SaveLoadManager.Instance.GetAchievementProgress(i, playerID);
+                        if (unlocked == true)
+                            achievement.achievementInfo.achievementStatus = AchievementStatus.Unlocked;
 
+                        achievementList.Add(achievement);
+                    }
+                }
+                else
+                {
+                    Debug.Log("[Achievement Manager] missing scriptable object reference");
+                }
+            }
         }
 
-        private void ScoreIncreased()
+        private void InvokeDefaultEvents()
         {
-            hiScoreController.CheckAchievement(UIManager.Instance.playerScore);
+            UIManager.Instance.ScoreIncreased += ScoreIncreased;
+            Enemy.EnemyManager.Instance.EnemyDestroyed += EnemyKilled;
+            Manager.GameManager.Instance.GameStarted += GamesPlayed;
         }
 
-        private void EnemyKilled()
+        private void ScoreIncreased(int playerID)
         {
-            enemiesKilledController.CheckAchievement(Enemy.EnemyManager.Instance.enemiesKilled);
+            CheckForAchievement(AchievementType.hiScore, UIManager.Instance.playerScore, playerID);
         }
 
-        private void GamesPlayed()
+        private void EnemyKilled(int playerID)
         {
-            gamesPlayedController.CheckAchievement(BTManager.GameManager.Instance.gamesPlayed);
+            CheckForAchievement(AchievementType.enemyKilled, Enemy.EnemyManager.Instance.enemiesKilled, playerID);
         }
 
-        public void CheckAchievement(AchievementType achievementType, int achievementIndex)
+        private void GamesPlayed()
         {
-            AchievementCheck?.Invoke(achievementType, achievementIndex);
+            CheckForAchievement(AchievementType.gamesPlayed, Manager.GameManager.Instance.gamesPlayed, 0);
         }
 
-        private void CallAchievementUI(AchievementType achievementType, int achievementIndex)
+        void CheckForAchievement(AchievementType achievementType, int achievedVal, int playerID)
         {
-            for (int i = 0; i < achievementScriptable.achievements.Count; i++)
+            for (int i = 0; i < achievementList.Count; i++)
             {
-                if(achievementScriptable.achievements[i].achievementType == achievementType)
+                if (CheckAchievementType(achievementType, i))
```

Comment:
it was probably more efficient to store the achievements as dict- so you can get quickly get achievements of a specific type for future ref